src/com/cdd/bao/template/SchemaVocab.java

```diff
@@ -91,7 +92,9 @@ public SchemaVocab(Vocabulary vocab, Schema[] templates)
 			termList[n].uri = termURI[n];
 			termList[n].label = vocab.getLabel(termURI[n]);
 			termList[n].descr = vocab.getDescr(termURI[n]);
-			termList[n].altLabels = vocab.getAltLabels(termURI[n]);
+			List<String> myList = new ArrayList<>(Arrays.asList(vocab.getAltLabels(termURI[n]))); // convert array of altLabels to a list
```

Comment:
You had the right idea with Vocabulary.java: keep altLabels as a primitive array, then use ArrayUtils.remove to delete an entry, if necessary.